gremlin-dotnet/src/Gremlin.Net/Driver/Connection.cs

```diff
@@ -114,11 +116,18 @@ internal class Connection : IConnection
                             result.Add(d);
                         }
                 }
+
+                if (status.Code == ResponseStatusCode.Success || status.Code == ResponseStatusCode.NoContent)
+                {
+                    statusAttributes = receivedMsg.Status.Attributes;
```

Comment:
Anything we expose should be agnostic to the serialization format. Status attributes should not be dependent on how GraphSON3 composes the response.

In this case, we are exposing a dictionary containing `@type` and `@value`, where `@value` is a `JArray` (a third party type).

Furthermore, we should hide any third party types from our API, see TINKERPOP-1696 for example.

Reply:
I do not think I am very clear on this. But , based on what I get, you want o strip down the @type and @value and build a internal map from @value, right ? If yes, I would to prefer to pass the wire message from the server and let the clients handle the message.